python/pyiceberg/catalog/glue.py

```diff
@@ -0,0 +1,247 @@
+        except self.glue.exceptions.EntityNotFoundException as e:
+            raise NoSuchNamespaceError(f"The database: {database_name} does not exist") from e
+
+        if database_location := response.get(PROP_GLUE_DATABASE).get(PROP_GLUE_DATABASE_LOCATION):
+            return f"{database_location}/{table_name}"
+
+        if PROP_WAREHOUSE in self.properties:
+            return f"{self.properties[PROP_WAREHOUSE]}/{database_name}.db/{table_name}"
+
+        raise ValueError("No default path is set, please specify a location when creating a table")
+
+    def _resolve_table_location(self, location: Optional[str], database_name: str, table_name: str) -> str:
+        if not location:
+            return self._default_warehouse_location(database_name, table_name)
+        return location
+
+    def create_table(
+        self,
+        identifier: Union[str, Identifier],
+        schema: Schema,
+        location: Optional[str] = None,
+        partition_spec: PartitionSpec = UNPARTITIONED_PARTITION_SPEC,
+        sort_order: SortOrder = UNSORTED_SORT_ORDER,
+        properties: Properties = EMPTY_DICT,
+    ) -> Table:
+        """Create an Iceberg table in Glue catalog
+        Args:
+            identifier: Table identifier.
+            schema: Table's schema.
+            location: Location for the table. Optional Argument.
+            partition_spec: PartitionSpec for the table.
+            sort_order: SortOrder for the table.
+            properties: Table properties that can be a string based dictionary.
+        Returns:
+            Table: the created table instance
+        Raises:
+            AlreadyExistsError: If a table with the name already exists
+            ValueError: If the identifier is invalid
+        """
+        database_name, table_name = self.identifier_to_tuple(identifier)
+
+        location = self._resolve_table_location(location, database_name, table_name)
+        metadata_location = f"{location}/metadata/00000-{uuid.uuid4()}.metadata.json"
+        metadata = new_table_metadata(
+            location=location, schema=schema, partition_spec=partition_spec, sort_order=sort_order, properties=properties
+        )
+        io = load_file_io({**self.properties, **properties}, location=location)
+        _write_metadata(metadata, io, metadata_location)
+        try:
+            self.glue.create_table(
+                DatabaseName=database_name, TableInput=_construct_table_input(table_name, metadata_location, properties)
+            )
+        except self.glue.exceptions.AlreadyExistsException as e:
+            raise TableAlreadyExistsError(f"Table {database_name}.{table_name} already exists") from e
+        except self.glue.exceptions.EntityNotFoundException as e:
+            raise NoSuchNamespaceError(f"Database {database_name} not found") from e
+
+        try:
+            load_table_response = self.glue.get_table(DatabaseName=database_name, Name=table_name)
```

Comment:
Can this call `load_table` instead of duplicating code?

Reply:
Thank you for your suggestion. I refactored it to take advantage of the `load_table` below